pkg/controllers/job/plugins/distributed-framework/mpi/mpi.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2022 The Volcano Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mpi
+
+import (
+	"flag"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/klog"
+
+	batch "volcano.sh/apis/pkg/apis/batch/v1alpha1"
+	pluginsinterface "volcano.sh/volcano/pkg/controllers/job/plugins/interface"
+)
+
+const (
+	MPIPluignName = "mpi"
+	defaultPort   = 22
+)
+
+type MpiPlugin struct {
+	MpiArguments []string
+	clientset    pluginsinterface.PluginClientset
+	MasterName   string
+	WorkerName   string
+	port         int
+}
+
+// New creates mpi plugin.
+func New(client pluginsinterface.PluginClientset, arguments []string) pluginsinterface.PluginInterface {
+	mp := MpiPlugin{MpiArguments: arguments, clientset: client}
```

Comment:
If struct `MpiPlugin` is available for instances out of this package, why is function `New` needed? Perhaps we can define it `mpiPlugin`.

Reply:
because I use  struct `MpiPlugin` in webhook pkg.
in pkg/webhooks/admission/jobs/plugins/mpi/mpi.go line 26
Is there a better way to handle it here?

Reply:
Got it. If that, `New` function is not so necessary.